### PR TITLE
[nrf noup] cmake: include nrf reports cmake from repo

### PR DESCRIPTION
**Pull request by hakonfam**
To make patches simpler and diff smaller.

nrf pr https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1741

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

### CODE REVIEW

**Comment by SebastianBoe** [comment]
DNM until the ongoing upmerge procedure is completed

File: cmake/reports/CMakeLists.txt [hunk 1]
```diff
@@ -17,12 +17,4 @@ foreach(report ram_report rom_report)
...
-add_custom_command(
-  TARGET rom_report
-  POST_BUILD
-  COMMAND
-  ${PYTHON_EXECUTABLE}
-  ${ZEPHYR_BASE}/../nrf/scripts/partition_manager_report.py
-  --input ${CMAKE_BINARY_DIR}/partitions.yml
-  "$<$<NOT:$<TARGET_EXISTS:partition_manager>>:--quiet>"
-  )
+include(${ZEPHYR_BASE}/../nrf/cmake/reports.cmake)
```
Comment (SebastianBoe) [inline cmake/reports/CMakeLists.txt]: I think we have a variable for ${ZEPHYR_BASE}/../nrf now

Comment (hakonfam) [reply cmake/reports/CMakeLists.txt]: I'm not able to find it. I copied what is used in boilerplate.cmake

Comment (SebastianBoe) [reply cmake/reports/CMakeLists.txt]: ZEPHYR_NRF_MODULE_DIR: /home/sebo/ncs/nrf

Comment (hakonfam) [reply cmake/reports/CMakeLists.txt]: THis will be made available after upmerge, will wait for this anyway.

File: cmake/reports/CMakeLists.txt [hunk 2]
```diff
@@ -14,5 +14,6 @@ foreach(report ram_report rom_report)
     --nm      ${CMAKE_NM}
     -o ${PROJECT_BINARY_DIR}
     DEPENDS ${logical_target_for_zephyr_elf}
+            $<TARGET_PROPERTY:zephyr_property_target,${report}_DEPENDENCIES>
```
Comment (tejlmand) [inline cmake/reports/CMakeLists.txt]: This comment refers to commit message, but placed here as I cannot comment directly i message:
```
 [nrf toup] cmake: allow projects to add report dependencies. 
```
should have been:
```
 [nrf fromlist] cmake: allow projects to add report dependencies. 
```
as we now have : https://github.com/zephyrproject-rtos/zephyr/pull/22006

Comment (hakonfam) [reply cmake/reports/CMakeLists.txt]: fixed

**Comment by mbolivar** [comment]
This repository's history has been rewritten for the v1.2 release. Please rebase this PR if still relevant, or close it if no longer needed. Thanks!

**Comment by hakonfam** [comment]
This could simply be a revert of the original commit adding these lines, but it looks like all the PM commits has been squashed, so I'm not sure how to do this in a way which simplifies the upmerge process. @mbolivar does this look OK?

**Comment by hakonfam** [comment]
Ping @SebastianBoe

**Comment by mbolivar** [comment]
>  @mbolivar does this look OK?

Looks great. I'll squash it in next rebase.

**Pull request merged**